pkg/apis/storage/types.go

```diff
@@ -157,7 +157,12 @@ type VolumeAttachmentSource struct {
 	// +optional
 	PersistentVolumeName *string
 
-	// Placeholder for *VolumeSource to accommodate inline volumes in pods.
+	// Translated VolumeSource from a pod to a CSIPersistentVolumeSource
+	// to support shimming of in-tree inline volumes to a CSI backend.
+	// This field is alpha-level and is only honored by servers that
+	// enable the CSIMigration feature.
+	// +optional
+	InlineVolumeSource *api.CSIPersistentVolumeSource
```

Comment:
Thinking forward:

Will we *ever* have a new storage spec after CSI?  Seems unlikely, but possible.  Will that have inline volumes?  Probably.  How could those be expressed here?  We have a CSI-specific field with a generic name.

Reply:
agree. something like `InlineCSIVolumeSource` leaves us wiggle room (and is more descriptive of the current field)